cmd/kubeadm/app/util/tokens.go

```diff
@@ -127,7 +127,10 @@ func DiscoveryPort(d *kubeadmapi.TokenDiscovery) int32 {
 // UpdateOrCreateToken attempts to update a token with the given ID, or create if it does
 // not already exist.
 func UpdateOrCreateToken(client *clientset.Clientset, d *kubeadmapi.TokenDiscovery, tokenDuration time.Duration) error {
-	secretName := fmt.Sprintf("%s%s", BootstrapTokenSecretPrefix, d.ID)
+	// Lower-case token ID because it will be used as part of a DNS label
+	// and in Kubernetes DNS labels must be lower-case.
+	// See /pkg/util/validation/validation.go.
+	secretName := strings.ToLower(fmt.Sprintf("%s%s", BootstrapTokenSecretPrefix, d.ID))
```

Comment:
@liggitt remove LGTM label and let's wait for luxas input or someone's else in the owners file. I don't want to complain to the users about being case-sensitive because this isn't clear to anyone but people directly using the API as we're doing here. The error message is not even clear.